src/coprocessor/Client.cc

```diff
@@ -123,19 +123,36 @@ std::vector<BatchCopTask> balanceBatchCopTasks(std::vector<BatchCopTask> && orig
     }
 
     // Only one tiflash store
-    if (original_tasks.size() <= 1)
+    if (original_tasks.size() <= 1 && !is_mpp)
```

Comment:
add some comments about why we can not early return when `is_mpp == true`.

I think it is because we need to detect whether the tiflash node is alive or not?

Reply:
I think for non-mpp mode, cannot early return either. I thinks it's a bug need to fix later.